server/controllers/stock/reports/stock/exit_report.js

```diff
@@ -0,0 +1,166 @@
+    })
+    .catch(next)
+    .done();
+}
+
+/**
+ * fetchDepotDetails
+ * @param {number} depotUuid depot uuid
+ */
+function fetchDepotDetails(depotUuid) {
+  const query = 'SELECT text FROM depot WHERE uuid = ?';
+  return db.one(query, [db.bid(depotUuid)]);
+}
+
+/**
+ * @function groupCollection
+ * @description group collected data by inventory
+ */
+function groupCollection(exitCollection) {
+  const collection = {};
+
+  // exit to patient
+  collection.exitToPatient = _.groupBy(exitCollection.exitToPatient, row => row.text);
+  collection.exitToPatient = _.map(collection.exitToPatient, formatExit);
+  collection.exitToPatientEmpty = (Object.keys(collection.exitToPatient).length === 0);
+
+  // exit to service
+  collection.exitToService = _.groupBy(exitCollection.exitToService, row => row.text);
+  collection.exitToService = _.map(collection.exitToService, formatExit);
+  collection.exitToServiceEmpty = (Object.keys(collection.exitToService).length === 0);
+
+  // exit to depot
+  collection.exitToDepot = _.groupBy(exitCollection.exitToDepot, row => row.text);
+  collection.exitToDepot = _.map(collection.exitToDepot, formatExit);
+  collection.exitToDepotEmpty = (Object.keys(collection.exitToDepot).length === 0);
+
+  // exit to loss
+  collection.exitToLoss = _.groupBy(exitCollection.exitToLoss, row => row.text);
+  collection.exitToLoss = _.map(collection.exitToLoss, formatExit);
+  collection.exitToLossEmpty = (Object.keys(collection.exitToLoss).length === 0);
+
+  return collection;
+}
+
+/**
+ * @function formatExit
+ */
+function formatExit(value, index) {
+  return {
+    inventory_name : index,
+    inventory_unit : value && value[0] ? value[0].unit_text : '',
+    inventory_stock_exit_data : value,
+    inventory_stock_exit_quantity : _.sumBy(value, 'quantity'),
+  };
+}
+
+/**
+ * @function collect
+ * @param {object} params query parameters
+ * @return {object} { exitToPatient: [], exitToService: [], exitToDepot: [], exitToLoss: [] }
+ */
+async function collect(params) {
+  const {
+    depotUuid,
+    dateFrom,
+    dateTo,
+    showDetails,
+    includePatientExit,
+    includeServiceExit,
+    includeDepotExit,
+    includeLossExit,
+  } = params;
+
+  const data = {};
+
+  try {
+    // get stock exit to patient
+    if (includePatientExit) {
+      data.exitToPatient = await StockExitToPatient.fetch(depotUuid, dateFrom, dateTo, showDetails);
+    }
+
+    // get stock exit to service
+    if (includeServiceExit) {
+      data.exitToService = await StockExitToService.fetch(depotUuid, dateFrom, dateTo, showDetails);
+    }
+
+    // get stock exit to other depot
+    if (includeDepotExit) {
+      data.exitToDepot = await StockExitToDepot.fetch(depotUuid, dateFrom, dateTo, showDetails);
+    }
+
+    // get stock exit to loss
+    if (includeLossExit) {
+      data.exitToLoss = await StockExitToLoss.fetch(depotUuid, dateFrom, dateTo, showDetails);
+    }
+
+  } catch (error) {
```

Comment:
Actually, there isn't a need for this `try/catch` block.  If any of the code throws an error, the whole function will throw an error and you can catch it in your top-level `try/catch` block.